compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -864,15 +864,13 @@ impl<'tcx> MirBorrowckCtxt<'_, 'tcx> {
         };
 
         let tcx = self.infcx.tcx;
-        let body_parent_did = tcx.opt_parent(self.mir_def_id().to_def_id())?;
-        if tcx.parent(region.def_id) != body_parent_did
-            || tcx.def_kind(body_parent_did) != DefKind::Impl
```

Comment:
Basically this check was too strict -- we don't care if the direct parent of the body is an impl, we only care that the region originates from the impl.